src/device.rs

```diff
@@ -409,8 +436,8 @@ pub trait DeviceV1_0 {
                                        pipelines.as_mut_ptr());
         pipelines.set_len(create_infos.len());
         match err_code {
-            vk::Result::Success => Ok(pipelines),
-            _ => Err(err_code),
+            vk::Result::Success => (pipelines, Ok(())),
+            _ => (pipelines, Err(err_code)),
```

Comment:
Good point there. After reading the linked page I would agree with you.
I mostly followed the `Multiple Pipeline Creation` section.
> If any pipeline fails creation (for example, due
to out of memory errors), the vkCreate*Pipelines commands will return an error code. The
implementation will attempt to create all pipelines, and only return VK_NULL_HANDLE values for
those that actually failed.

The specification isn't really clear here I think.

Therefore, I tried a bit around to get one of these mysterious `VK_NULL_HANDLES`. Setting `s_type` in the `GraphicsPipelineCreateInfo` to `vk::StructureType::ComputePipelineCreateInfo` results in a `Err(ErrorInitializationFailed)` on my AMD system, which isn't listed as a return code!
Creating two pipelines (one of them failed) gave me this as return code:
`([0x0, 0x1edbeea5830], Err(ErrorInitializationFailed))`
So one null handle, a valid pipeline handle and an error.

I'm not sure if this is correct behavior from the driver or the specs are missing something, but imo the proposed code change handles this scenario more gracefully.